teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,473 @@
+
+A TEP can be created with the `implementable` state if it doesn't need
+any more discussion and got approved as it.
+
+See [Examples](#examples) to see examples of TEP workflow on use cases.
+
+### Git Implementation
+
+TEPs are checked into the community repo under the `/teps` directory.
+
+New TEPs can be checked in with a file name in the form of
+`draft-YYYYMMDD-my-title.md`. As significant work is done on the TEP,
+the authors can assign a TEP number. No other changes should be put in
+that PR so that it can be approved quickly and minimize merge
+conflicts. The TEP number can also be done as part of the initial
+submission if the PR is likely to be uncontested and merged quickly.
+
+### Prior Art
+
+The TEP process as proposed was essentially adapted from the
+[Kubernetes KEP process][], which itself is essentially stolen from the [Rust RFC
+process][] which itself seems to be very similar to the [Python PEP
+process][]
+
+[Rust RFC process]: https://github.com/rust-lang/rfcs
+[Kubernetes KEP process]: https://github.com/kubernetes/enhancements/tree/master/keps
+[Python PEP process]: https://www.python.org/dev/peps/
+
+
+## Examples
+
+Let's give some example of workflow to give reader a better
+understanding on how and when TEP should be created and how they are
+managed across time.
+
+Those are examples, and do not necessarily reflect what happened, or
+will happens on the particular subject they are about. They are here
+to give more context and ideas on different situation that could rise
+while following the TEP process.
+
+### Share Task and Pipeline as OCI artifact
+
+For more context, this is linked to the following:
+
+- [Feature: Versioning on Tasks/Pipelines](https://github.com/tektoncd/pipeline/issues/1839)
+- [Oci tool: makes use of oci-artifacts to store and retrieve Tekton resources](https://github.com/tektoncd/experimental/pull/461)
+
+1. An initial design doc is crafted (let's imagine it is [Tekton OCI
+Image
+Catalog](https://docs.google.com/document/d/1zUVrIbGZh2R9dawKQ9Hm1Cx3GevKIfOcRO3fFLdmBDc/edit#heading=h.tp9mko2koenr)).
+   An experimental project has already been created and a
+   proof-of-concept demoed during a working group. The next step is to
+   create a TEP (and continue work on the proof-of-concept if need be).
+2. From this design docs, a TEP is created with the content of the
+   design document.
+3. It is approved with a `proposed` state, which means :
+   - We acknowledge this is important for the project, and needs to be
+     worked on
+   - It needs some work and discussion based on the initial proposal
+4. The TEP is being disscussed during Working Group(s) — it can be the
+   main one, a specific one, the API Working Group.
+
+   During those discussion it is clear that some work needs to be
+   done:
+   - Define a Spec for the OCI image (layers, metadata, configuration)
+     The experimental project can be used to demo and validate that spec.
+   - Once the spec is agreed on
```

Comment:
this bullet is kinda odd on its own, maybe the next one can be combined with it?

Reply:
Oh :scream_cat: The rest of the sentence might be missing :sweat: